polygon/heimdall/entity_fetcher.go

```diff
@@ -96,7 +96,7 @@ func (f *entityFetcherImpl[TEntity]) FetchEntitiesRange(ctx context.Context, idR
 		if err != nil {
 			return nil, err
 		}
-		startIndex := idRange.Start - 1
+		startIndex := idRange.Start
```

Comment:
spans start at 0 (ie we have span0 - https://heimdall-api-amoy.polygon.technology/bor/span/0) 
checkpoints start at 1  (ie we do not have https://heimdall-api-amoy.polygon.technology/checkpoints/0 but have https://heimdall-api-amoy.polygon.technology/checkpoints/1)

looks like we need to introduce a `fetchAllEntitiesIdxOffset` attribute to the `entityFetcher`, set it to 0 for spans, and 1 for checkpoints then this becomes `startIndex := idRange.Start - f. fetchAllEntitiesIdxOffset`